### ISSUE TITLE
Unused arguments in lambda should not have types

### ISSUE THREAD

**Issue by akuleshov7** [analysis-dev/diktat#1172]
This is a great stylistic bug, that is not handled by IDEA: 
```
foo {_, _: Int, a: Int - > println("hi") }
```

diktat should detect and remove a type of such anonymous arguments (`_: Int`)

**Comment by petertrr** [comment]
What if there are several overloads of function `foo` with the same number of arguments but different types? Then type of the lambda would be ambiguous. I guess it would make sense for named arguments, because I can't imagine a case, when its type cannot be inferred correctly.

**Comment by akuleshov7** [comment]
>>>  I guess it would make sense for named arguments
I meant only unnamed arguments

**Comment by petertrr** [comment]
I meant some weird corner cases like
```kotlin
fun <T> foo(f: (a: T, b: String) -> Unit) = Unit
fun bar() {
    foo { _: Int, s -> println(s) }
}
```
Here removing `: Int` will cause compilation failure

**Comment by ephemient** [comment]
To give another example on built-in stdlib functions,

```kotlin
list.fold(null) { _: Any?, _ -> }
```

will not type-check if `_: Any?` is replaced by `_`.

**Comment by akuleshov7** [comment]
Hm, yes, that's true @petertrr @ephemient. But we need to cover at least simple cases somehow...

**Comment by ephemient** [comment]
Without full type inference it is not possible to determine whether removing the explicit type keeps the same behavior of code or not.
